alibi/confidence/model_linearity.py

```diff
@@ -84,23 +82,19 @@ def _calculate_global_linearity(predict_fn: Callable, input_shape: Tuple, X_samp
 
     if model_type == 'classifier':
         out_sum = np.log(predict_fn(summ) + 1e-10)
-        out_sum_shape = out_sum.shape[1:]
     elif model_type == 'regressor':
         out_sum = predict_fn(summ)
         out_sum_shape = out_sum.shape[1:]
-        if len(out_sum.shape) == 1:
-            out_sum = out_sum.reshape((ss[0],) + (1,))
-        else:
-            out_sum = out_sum.reshape((ss[0],) + out_sum_shape)
+        out_sum = out_sum.reshape((ss[0],) + (1,)) if len(out_sum.shape) == 1 else out_sum.reshape(
+            (ss[0],) + out_sum_shape)
```

Comment:
Not 100% convinced this one is actually more readable, but it is quite subjective I guess...